core/libs/shared/src/tree_like.rs

```diff
@@ -77,6 +100,19 @@ where
     }
 }
 
+impl<T> TreeLikeMut for &mut T
```

Comment:
new blanket implementation makes a mutable reference to any `TreeLikeMut` also `TreeLikeMut`